### PR TITLE
WT-13478 Improve Block manager test consistency

### PR DESCRIPTION
**Pull request by jeremythorp**
This ticket and PR improve the consistency of the Block manager Catch2-based tests. In particular, tests have been refactored and helper code moved to be clearer to understand and make better use of Catch2.

Changes make by this PR:
- Moved shared block manager test utility functions into `block_util.cpp`
- Separated TEST_CASE for __wti_block_ext_prealloc into two TEST_CASES to simplify test setup/cleanup
- Refactored extent list tests to use common setup/cleanup where appropriate
- Changed `operator()(const off_size &left, const off_size &right)` to be `const` as it can be.
- Tidied up constness for function parameters.

### CODE REVIEW

File: test/unittest/tests/block/test_extent_list_insert_wo_block.cpp [hunk 1]
```diff
@@ -33,12 +33,12 @@ TEST_CASE("Extent Lists: block_ext_insert", "[extent_list]")
...
     std::vector<WT_EXT **> stack(WT_SKIP_MAXDEPTH, nullptr);
 
+    /* Common setup. */
+    WT_EXTLIST extlist = {};
+
     SECTION("insert into an empty list has one element")
     {
         BREAK;
-        /* Setup. */
-        WT_EXTLIST extlist = {};
```
Comment (tod-johnson-mongodb) [inline test/unittest/tests/block/test_extent_list_insert_wo_block.cpp:40]: This change removes the independence of the SECTIONs within the TEST_CASE.
* Are "Common setup" and "Common cleanup" both run for each SECTION, or just once before the first SECTION and once after the last SECTION? If the first, this change is desirable. If the later, this change is not desirable.

**Review by tod-johnson-mongodb — COMMENTED** [review]
I read all of this. I don't know whether to approve this or not since I don't know what moving setup and cleanup out of SECTION into TEST_CASE does. See my comment.

**Comment by jeremythorp** [comment]
Hi @tod-johnson-mongodb ,

The [Catch2 documentation states](https://github.com/catchorg/Catch2/blob/devel/docs/tutorial.md#test-cases-and-sections) (and has a great example):

> For each SECTION the TEST_CASE is executed from the start.

So a developer can write common setup and cleanup code in a TEST_CASE (but outside a SECTION) and then that code will be executed for each and every SECTION.

**Review by jiechenbo — APPROVED** [review]
LGTM! @jeremythorp is it worth looking at the labels? Should we bunch them together?

**Comment by jeremythorp** [comment]
> LGTM! @jeremythorp is it worth looking at the labels? Should we bunch them together?

Hi @jiechenbo , good suggestion. Most of the block manager TES_CASE labels were already consistent, but I've now addressed the rest. Let me know what you think. Thanks, Jeremy

**Review by tod-johnson-mongodb — APPROVED** [review]
LGTM

**Pull request merged**